### PR TITLE
Replace Woopay’s Development Environment Constants With Global Variables

### PR DESCRIPTION
**Pull request by lovo-h**
Fixes 2740-gh-Automattic/woopay

#### Changes proposed in this Pull Request

The changes in this PR ensure that the development environment uses global variables as opposed to hard-coded constants.

#### Testing instructions

See testing instructions in  2788-gh-Automattic/woopay.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `9019` or branch name `debt/2740-woopay-remove-hard-coded-secrets` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=debt/2740-woopay-remove-hard-coded-secrets) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: b408a142e71291c760ebe09602d2650d110d2f5a
- Build time: 2024-07-23 00:38:12 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: includes/woopay/class-woopay-utilities.php [hunk 1]
```diff
@@ -252,14 +252,35 @@ public static function get_woopay_url() {
...
+			// Using WooPay production: Use the blog token secret from the store blog.
+			return Jetpack_Options::get_option( 'blog_token' );
+		} elseif ( apply_filters( 'wcpay_woopay_use_blog_token', false ) ) {
+			// Requested to use the blog token secret from the store blog.
+			return Jetpack_Options::get_option( 'blog_token' );
+		} elseif ( defined( 'DEV_BLOG_TOKEN_SECRET' ) ) {
+			// Has a defined dev blog token secret: Use it.
+			return DEV_BLOG_TOKEN_SECRET;
+		} else {
+			// TODO: Should we log this?
```
Comment (cesarcosta99) [inline includes/woopay/class-woopay-utilities.php]: I think we should definitely log this 👍

**Review by cesarcosta99 — COMMENTED** [review]
Code looks good! I left some questions in the other PR. For now, my suggestion is to address the TODO.

**Review by cesarcosta99 — APPROVED** [review]
LGTM :shipit:

- https://github.com/Automattic/woopay/issues/2740 testing instructions ✅ 
- `npm run test:php` ✅

**Pull request merged**